consensus/hotstuff/notifications/pubsub/qc_created_distributor.go

```diff
@@ -0,0 +1,103 @@
+package pubsub
+
+import (
+	"sync"
+
+	"github.com/onflow/flow-go/consensus/hotstuff"
+	"github.com/onflow/flow-go/consensus/hotstuff/model"
+	"github.com/onflow/flow-go/model/flow"
+)
+
+type OnQCCreatedConsumer = func(qc *flow.QuorumCertificate)
+
+// QCCreatedDistributor subscribes for qc created event from hotstuff and distributes it to subscribers
+// Objects are concurrency safe.
+// NOTE: it can be refactored to work without lock since usually we never subscribe after startup. Mostly
+// list of observers is static.
+type QCCreatedDistributor struct {
+	qcCreatedConsumers []OnQCCreatedConsumer
+	lock               sync.RWMutex
+}
+
+var _ hotstuff.Consumer = (*QCCreatedDistributor)(nil)
```

Comment:
I would suggest to create a dedicated interface for this (well, specifically for _every_ data producer). Otherwise, it will be really confusing to determine later which component produces what data.

Reply:
Done